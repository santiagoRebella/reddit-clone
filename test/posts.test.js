const React = require('react');
const immutable = require('immutable');
const Posts = require('components/posts');
const renderer = require('react-test-renderer');

test('Posts is rendered correctly', () => {
  const component = renderer.create(
    <Posts
        list={immutable.List([{
          data: {
            id: "someid",
            title: "Test title",
            description: "Test Description",
            url: "Test Url",
            image: "Test image",
            author: "Test author",
            time: 1234567
          }}, {
          data: {
            id: "someotherid",
            title: "other Test title",
            description: "other Test Description",
            url: "other Test Url",
            image: "other Test image",
            author: "other Test author",
            time: 1111123
          }}, {
          data: {
            id: "anid",
            title: "A Test title",
            description: "A Test Description",
            url: "A Test Url",
            image: "A Test image",
            author: "A Test author",
            time: 123452367
          }}
        ])}
        subreddit="someSubreddit"
        onViewChange={jest.fn()}
        fetchPosts={jest.fn()}

    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

