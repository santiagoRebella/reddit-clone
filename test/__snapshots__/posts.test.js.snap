// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts is rendered correctly 1`] = `
<section
  className="posts"
>
  <div
    className="card"
    style={
      Object {
        "display": "inline-block",
        "margin": 10,
        "width": 240,
      }
    }
  >
    <img
      alt="Post image cap"
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <h6
        className="card-title"
      >
        author: 
        Test author
      </h6>
      <p
        className="card-text"
      >
        created: 
        Invalid date
      </p>
      <h5
        className="card-title"
      >
        Test title
      </h5>
      <p
        className="card-text"
      >
        Test Description
      </p>
      <a
        className="btn btn-primary"
        href="https://reddit.comundefined"
      >
        Comments
      </a>
    </div>
  </div>
  <div
    className="card"
    style={
      Object {
        "display": "inline-block",
        "margin": 10,
        "width": 240,
      }
    }
  >
    <img
      alt="Post image cap"
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <h6
        className="card-title"
      >
        author: 
        other Test author
      </h6>
      <p
        className="card-text"
      >
        created: 
        Invalid date
      </p>
      <h5
        className="card-title"
      >
        other Test title
      </h5>
      <p
        className="card-text"
      >
        other Test Description
      </p>
      <a
        className="btn btn-primary"
        href="https://reddit.comundefined"
      >
        Comments
      </a>
    </div>
  </div>
  <div
    className="card"
    style={
      Object {
        "display": "inline-block",
        "margin": 10,
        "width": 240,
      }
    }
  >
    <img
      alt="Post image cap"
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <h6
        className="card-title"
      >
        author: 
        A Test author
      </h6>
      <p
        className="card-text"
      >
        created: 
        Invalid date
      </p>
      <h5
        className="card-title"
      >
        A Test title
      </h5>
      <p
        className="card-text"
      >
        A Test Description
      </p>
      <a
        className="btn btn-primary"
        href="https://reddit.comundefined"
      >
        Comments
      </a>
    </div>
  </div>
</section>
`;
